plugins/inputs/ethtool/ethtool_linux.go

```diff
@@ -4,180 +4,134 @@
 package ethtool
 
 import (
-	"net"
-	"regexp"
-	"strings"
-	"sync"
-
-	"github.com/pkg/errors"
-	ethtoolLib "github.com/safchain/ethtool"
+	"os"
+	"path/filepath"
 
 	"github.com/influxdata/telegraf"
-	"github.com/influxdata/telegraf/filter"
 	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/vishvananda/netns"
 )
 
 type CommandEthtool struct {
-	ethtool *ethtoolLib.Ethtool
-}
-
-func (e *Ethtool) Gather(acc telegraf.Accumulator) error {
```

Comment:
Hi @zeffron I was under the impression ethtool only exists for linux. Are there alternate implementations for the other platforms Telegraf supports (Windows, MacOS, FreeBSD)?

This function and a few others were moved to the platform-independent ethtool.go. It doesn't make sense to move the plugin implementation there if ethtool only exists on linux.

Reply:
Ethtool only exists on Linux (at least for now).

Because (some of) these functions are part of the interface, they had to have stubs defined in the non-Linux file. They didn't actually use any of the ethtool (or namespace) functionality, though, calling out through abstractions or just performing logic. Since that logic isn't platform dependent, having separate functions for Linux and non-Linux creates opportunities for divergence. It also prevented unit tests from running on non-Linux platforms, and since unit tests don't rely on ethtool or Linux namespaces, either (because of mocking), that just created a development burden. See https://github.com/influxdata/telegraf/pull/11895#discussion_r997558646

Moving the logic that was platform independent to the platform independent file solved the unit test need.

Reply:
Not sure which changes resulted in unit tests failing on macOS again, but they are because `Ethtool.Init()` panics on nil `Log`, so I'm putting back the restriction.

Reply:
Ah, what changed is that the tests now call `Ethtool.Init()` in the Gather tests after the down interface merge.

Reply:
@zeffron I can understand why abstracting out has made development easier, however we hesitate to accept this now because we prefer for plugins to only build and run on the platforms that they support. Having tests randomly fail on macOS because of a plugin that macOS doesn't even support is something we would like to avoid. More importantly, we have #11689 that we would like to land as a part of our major release that uses these tags even further to omit plugins not supported on the build platform.

Reply:
@powersj 
1) I've already changed it back so tests only run on Linux.

Regarding the #11689, I'm not understanding what that has to do with this? That file moved everything into the main file, so I don't see how my moving functions into the main file logically conflicts? (Obviously there's going to be a merge conflict in Git, but that would happen even if I left the functions where there were originally.) If you really want I can move these functions back to the Linux only file, though.

Reply:
@powersj 
1. I've already changed it back so tests only run on Linux.
2. Regarding the #11689, I'm not understanding what that has to do with this? That file moved everything into the main file, so I don't see how my moving functions into the main file logically conflicts? (Obviously there's going to be a merge conflict in Git, but that would happen even if I left the functions where there were originally.) If you really want I can move these functions back to the Linux only file, though.

Reply:
Functions have been put back to their original files.